### ISSUE TITLE
Chain+asset selection component

### ISSUE THREAD

**Issue by alexwhte** [connext/monorepo#4670]
## Problem
When users are performing any to any transfers (any assets from any chain), it makes the most sense to combine the asset selection and chain selection into a single click 

## Impact
We building a demo for chain abstraction which is urgent to demonstrate the UX to developers. So far this has had a strong reception
We are also looking at adding this to our bridge sometime this quarter https://github.com/connext/monorepo/issues/4149
This is a key piece of explaining chain abstracted UX and reducing friction in the UI 
 
## Proposed Solution
Build a component that is designed for our Bridge UI but that is isolate enough that we can use the code in our pending demo token selection here https://chain-abstraction-reference.vercel.app/

Basically, create the selection but please write the code it in a way that is reusable enough across the two UIs 

Please see examples  
- https://playground.mean.finance/create/137 
- https://app.joinwido.com/?utm_source=joinwidocom&utm_content=herobtn

## Acceptance Criteria
_From an end user perspective, the below criteria must be met to consider this done:_
- [ ] User can see all assets across all chains sorted by value
- [ ] Each asset has an asset main icon + chain sub-icon
- [ ] User can select any of these
- [ ] Their selection will have some sort of result that can be easily used to set the users's origin chain/network AND asset

![image](https://github.com/connext/monorepo/assets/37959170/f0cbe114-4d1c-49a1-84d7-20fe7ae75e87)

**Comment by sanchaymittal** [comment]
looks good to me!!

**Comment by sanchaymittal** [comment]
the component has been added to the demo app, further feedback will have new tasks.
